opendcs-rest-api/src/main/java/org/opendcs/odcsapi/opendcs_dep/TestDecoder.java

```diff
@@ -339,5 +343,33 @@ private static void decodedMsg2api(ApiDecodedMessage adm, DecodedMessage dm, Arr
 		}
 	}
 
+	/**
+	 * Synchronized to make sure multiple concurrent sessions don't init at the same time.
+	 * Also, only initialize once per minute.
+	 */
+	private static synchronized void initDecodes(OpenDcsDatabase db)
+			throws DbException
+	{
+		// Don't init more than once per minute.
+		long now = System.currentTimeMillis();
+		if (now - lastDecodesInitMsec < 60*1000L)
+			return;
+		lastDecodesInitMsec = now;
+
+		if (decodes.db.Database.getDb() == null)
```

Comment:
FYI Instances of OpenDcsDatabase are required to have a valid "Database" instance if still required, but our main intent is to move downstream code to the getDao feature and hide the Database detail, if still needed, in the getDao implementation.